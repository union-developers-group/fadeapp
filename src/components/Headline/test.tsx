import { render } from '@testing-library/react'
import { Headline } from '.'


describe('<Headline />', () => {
  it('should render by default', () => {
    const { getByRole, queryByTestId } = render(
      <Headline textPosition="center" as="h2" title="Hello World" />,
    )
    const heading = getByRole('heading', { name: 'Hello World' })
    const line = queryByTestId('underline')

    expect(heading.tagName.toLocaleLowerCase()).toBe('h2')
    expect(heading).toBeInTheDocument()
    expect(line).toBe(null)
  })

  it('should render custom heading', () => {
    const { getByRole } = render(
      <Headline textPosition="center" as="h1" title="Texto" />,
    )
    const heading = getByRole('heading', { name: 'Texto' })

    expect(heading.tagName.toLocaleLowerCase()).toBe('h1')
    expect(heading).toBeInTheDocument()
  })

  it('should render a line under the title', () => {
    const { queryByTestId } = render(
      <Headline textPosition="center" withLine title="Hello World" />,
    )
    const line = queryByTestId('underline')

    expect(line).not.toBe(null)
  })

  it('should match snapshot', () => {
    const { container } = render(
      <Headline textPosition="center" title="Hello" />,
    )

    expect(container).toMatchSnapshot()
  })
})
