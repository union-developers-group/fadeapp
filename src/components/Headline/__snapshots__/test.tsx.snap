// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Headline /> should match snapshot 1`] = `
<div>
  <div
    class="
relative flex flex-col text-white 
  min-w-[25.25rem] 
items-center"
  >
    <h2
      class="
  text-2xl font-bold sm:text-6xl  
} "
    >
      Hello
    </h2>
  </div>
</div>
`;
